### ISSUE TITLE
On Android uport signer method showSeed doesn't return anything

### ISSUE THREAD

**Issue by MarcoLude** [uport-project/react-native-uport-signer#41]
Hi all,

the following issues seem to appear only on Android.

We have a project with react native 0.59.9 and we are using react-native-uport-signer (1.5.0) to handle private key and seed phrase.
We create the seed by using "RNUportHDSigner.createSeed('prompt')"

We faced 2 weird issues while using the function "RNUportHDSigner.showSeed(uportAddress)".
1) If the smartphone has the fingerPrint hardware but the user has just set up a pin/password/swipe security method without the fingerprint, the showSeed method asks for the set security option (pin/password/swipe), but as soon as the user verifies it the showSeed method doesn't return anything. We cannot even catch any error, it seems just to go in an infinite loop. We use the following methodology:

> RNUportHDSigner.showSeed(uportAddress)
>    .then(seedPhrase=>{ //handle seed phrase })
>    .catch(error=>{console.log("error: ",error}) 

2) If we try to add a customized message to be prompt to the user, it doesn't work.
We use this:

> RNUportHDSigner.showSeed(uportAddress, 'Please, unlock your phone to reveal the recovery phrase')

But the message prompt is just "uPort".
Tested on Android 9, Android 7, Android 6 with multiple devices/emulators.

You can reproduce the issues by using also an emulator:
1) create a new react-native project using version <=0.59.9 and add react-native-uport-signer 1.5.0.
2) run the project on a smartphone/emulator that has the fingerprint hardware, but not a fingerprint registered. Use a pin/password/swipe instead.
3) create a seed by using RNUportHDSigner.createSeed('prompt')
4) show the seed by using RNUportHDSigner.showSeed(uportAddress)
5) try to log the seedPhrase returned by "showSeed"

The first one is an issue a bit urgent, could you please check it as soon as possible?

Thank you very much

**Comment by mirceanis** [comment]
Thanks for reporting this.
Do you have a project already setup that shows this behavior?

Probably unrelated but worth mentioning; are you seeing the same behavior with the latest version 1.5.2 ?

Also, in your reproduction steps you don't mention if the fingerprint has been setup or not, can you clarify?

**Comment by MarcoLude** [comment]
Hi Mircea,

you can try it on your demo app: https://github.com/uport-project/react-native-signer-demo.git

Use one android device/emulator that has the fingerprint hardware, but without setting up the fingerprint. Just set up a Pin/password/pattern. The ShowSeed, after creating the keypair, seems to not return anything

edit:
Yes, the same behaviour on 1.5.2

**Comment by mirceanis** [comment]
@MarcoLude I can't reproduce it on the demo project (RN 0.59.8, and using version 1.5.2 of this library).
I used emulators with both pin and pattern
`HD signer` -> `create keypair` -> `show seed`
Keyguard displays the prompt message "reveal the backup phrase"
After typing the PIN, keyguard disappears and the phrase is shown under the `show seed` button.

**Comment by mirceanis** [comment]
@MarcoLude can you provide a screen capture of your device showing the behavior you describe? Perhaps there's something I'm missing.
Otherwise we should close this.

**Comment by MarcoLude** [comment]
Hi Mircea,

these are 2 GIFs of uPort signer demo App (it shows a warning, on our project doesn't, so maybe it is more helpful)

One recorded on a real device Honor8 with Android 7 with setup the Pin and not the fingerprint.
One recorded on a real device Xiaomi Readme Note 3 with Android 6 with setup the Pattern and not the fingerprint.
Both devices have fingerprint hardware.

Honor8 (sadly it doesn't record the OS pin screen, but you can see the keyboard typing the pin)
![uport_Honor8](https://user-images.githubusercontent.com/38657984/66048440-d2871a00-e529-11e9-86dc-8a2fcbd56c83.gif)

XiaomiReadmeNote3
![uport_XiaomiNote3](https://user-images.githubusercontent.com/38657984/66048462-de72dc00-e529-11e9-8d6c-6cfe5e2f8f1f.gif)

We could not manage to make the demo app work on an Android 9, but our project, which is using this library, presents the same issue on a Samsung S10 with Android 9

**Comment by mirceanis** [comment]
Unfortunately, I don't have access to these devices, so I can't test. You will have to help with debugging information.

1. Does the same behavior happen when you use `single_prompt` instead of `prompt` when creating a seed?
2. What about signing jwt or transaction?
3. Is the emulator behaving correctly for you? If you are not seeing the same behavior I described earlier, then there's something else going on.

The mechanism that triggers the keyguard is the exact same one in all cases.

The trigger for the keyguard is done by the android keystore and the aspect of the keyguard is decided by the manufacturer. Besides the `prompt` message, there is no control over the aspect from the library.

I see that there are in fact errors thrown, and they look different on the 2 GIFs, it would probably help if you could paste the error messages here. It's hard to follow the GIFs.

Are you able to build and run pure android projects?
The android side of this react-native library is based on https://github.com/uport-project/uport-android-signer
That project has a demoapp that can be run from Android Studio.
This issue is probably unrelated to react-native so that project should show similar errors when following the fingerprint flows, but if it does not show errors, then there's something more tricky going on. It helps to narrow it down.

**Comment by MarcoLude** [comment]
I will answer point by point, I'm using the demo app:

1) Changing "prompt" with "single_prompt" changes the behavior. It doesn't ask for Pin and shows the seed phrase without prompting anything (both on emulator Nexus 5 with Android 7 and on real device Honor8 with Android 7).
2) "showSeed", "sign JWT" and "sign transaction" behave the same: the error is prompt.
3) the emulator Nexus 5 with Android 7 presents not always the same behavior: sometimes it works, sometimes it doesn't. 
This is an example screen with the caught warning on emulator
![Screenshot 2019-10-03 at 10 49 32](https://user-images.githubusercontent.com/38657984/66115656-90b1ae80-e5d1-11e9-88ea-e8fd755b0073.png)

The error when it's not caught:

![20191003-105139](https://user-images.githubusercontent.com/38657984/66116014-4b41b100-e5d2-11e9-9434-597c9297fa41.png)

The error when it's caught:

![Screenshot_20191003-104713](https://user-images.githubusercontent.com/38657984/66115953-2ea57900-e5d2-11e9-82ee-821cc8f01a47.png)


4) I've built and run the Android Project, and it works on both emulator and real device with the "KeyProtection.Level.PROMPT" with the same conditions.
